src/authorizer.ts

```diff
@@ -1,20 +1,23 @@
-import { LoaderFunction, redirect, json } from "@remix-run/server-runtime";
+import type { DataFunctionArgs } from "@remix-run/server-runtime";
+import { json, redirect } from "@remix-run/server-runtime";
 import { Authenticator } from "./authenticator";
 
-type LoaderArgs = Parameters<LoaderFunction>[0];
-
-type AuthorizeArgs<User, Data> = Omit<RuleContext<User, Data>, "user">;
+/**
+ * Extra data passed to the Authorizer from the loader or action
+ */
+type AuthorizeArgs<Data> = { data?: Data } & Omit<
+  DataFunctionArgs,
+  "context"
+> & {
+    context?: DataFunctionArgs["context"];
+  };
```

Comment:
Context shouldn't be option, I know you may not use it, and it's the most common thing, but the type is not optional in LoaderArgs so it shouldn't be here, or at least not the types the rules receives, we could accept context as optional but pass it to the rules with a default empty object.

Reply:
Makes sense. I've updated it to be optional in the call to `authorize` but the rule functions will always receive a context (an empty object if not provided).